sdks/python/apache_beam/runners/common.py

```diff
@@ -377,10 +377,6 @@ def __init__(self, output_processor, signature, context,
     input_args = input_args if input_args else []
     input_kwargs = input_kwargs if input_kwargs else {}
 
-    if not self.has_windowed_inputs:
```

Comment:
This is an important optimization for batch pipelines that use side inputs (e.g the TFX stuff). I see now how we were requesting side inputs in start(), but they're not window dependent. Perhaps we could defer this optimization to the first element that is processed. (At least a TODO would be in order.)

Reply:
Like you suggest, checking for at least one element or loading on first element should work, just trying to get everything working E2E before I try to improve this optimization.